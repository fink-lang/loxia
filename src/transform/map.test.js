import {fink2js} from '../testing';


test('compiles map', ()=> {
  expect(
    fink2js(`
      map item: item * 2
      map item=123: item * 2
      map {item}: item * 2
      map [x, y]: x + y

      flat_map item: [spam + item]

      flat_map item: [spam]
    `)
  ).toMatchSnapshot();
});


