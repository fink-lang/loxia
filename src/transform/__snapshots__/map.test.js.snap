// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`compiles map 1`] = `
"(function* map($items1) {
  for (const item of $items1) {
    const $result2 = item * 2;
    yield $result2;
  }
});

(function* map($items3) {
  for (const item = 123 of $items3) {
    const $result4 = item * 2;
    yield $result4;
  }
});

(function* map($items5) {
  for (const {
    item
  } of $items5) {
    const $result6 = item * 2;
    yield $result6;
  }
});

(function* map($items7) {
  for (const [x, y] of $items7) {
    const $result8 = x + y;
    yield $result8;
  }
});

(function* map($items9) {
  for (const item of $items9) {
    const $result10 = [spam + item];
    yield* $result10;
  }
});

(function* map($items11) {
  for (const item of $items11) {
    const $result12 = [spam];
    yield* $result12;
  }
});"
`;
